collectors/HostSystemCollector.py

```diff
@@ -21,17 +21,36 @@ def collect(self):
 
         g = GaugeMetricFamily('vrops_hostsystem', 'testtext', labels=['datacenter', 'cluster', 'hostsystem', 'statkey'])
 
-        for hs in self.get_hosts():
-            target = self.hosts[hs]['target']
-            for statkey in self.statkey_yaml["HostSystemCollector"]:
-                r = Resources()
-                if os.environ['DEBUG'] >= '2':
-                    print("looking at " + str(self.hosts[hs]))
-                value = r.get_latest_stat(target=target, token=self.hosts[hs]['token'], uuid=self.hosts[hs]['uuid'], key=statkey["statkey"])
-                if value is None:
-                    value = "0"
-                if os.environ['DEBUG'] >= '1':
-                    print(self.hosts[hs]['name'], "--add statkey:", statkey["label"], str(value))
-                g.add_metric(labels=[self.hosts[hs]['datacenter'], self.hosts[hs]['parent_cluster_name'],
-                                     self.hosts[hs]['name'], statkey["label"]], value=value)
+        #make one big request per stat id with all resource id's in its belly
+        for target in self.get_hosts_by_target():
+            token = self.get_target_tokens()
+            token = token[target]
+            if not token:
+                print("skipping " + target + " in HostSystemCollector, no token")
+
+            uuids = self.target_hosts[target]
+            for statkey_pair in self.statkey_yaml["HostSystemCollector"]:
+                statkey_label = statkey_pair['label']
+                statkey = statkey_pair['statkey']
+                values = Resources.get_latest_stat_multiple(target, token, uuids, statkey)
+                if not values:
+                    print("skipping statkey " + str(statkey) + " in HostSystemCollector, no return")
+                    continue
```

Comment:
how can it happen that a statkey does not exist? it is more likely that a uuid of a resource cannot be found.

Reply:
Thought behind is to catch, if something goes wrong when issuing the API command. When an UUID is missing, it still will end up reporting 200, just tried it. If we would query for statkey, which does not exist, you would immediately see that. It wouldn't happen for tested code, but it will happen for others, specifying non-existent keys during delevopment.

Reply:
O.K. Sure thing